test/TransitivePathTest.cpp

```diff
@@ -144,14 +213,141 @@ TEST(TransitivePathTest, unlimitedMaxLength) {
 
   TransitivePathSide left(std::nullopt, 0, Variable{"?start"}, 0);
   TransitivePathSide right(std::nullopt, 1, Variable{"?target"}, 1);
-  TransitivePath T(getQec(), nullptr, left, right, 1,
-                   std::numeric_limits<size_t>::max());
+  auto T =
+      makePathUnbound(std::move(sub), {Variable{"?start"}, Variable{"?target"}},
+                      left, right, 1, std::numeric_limits<size_t>::max());
+
+  auto resultTable = T->computeResultOnlyForTesting();
+  assertSameUnorderedContent(expected, resultTable.idTable());
+}
+
+TEST_P(TransitivePathTest, idToLeftBound) {
+  IdTable sub(2, makeAllocator());
+  sub.push_back({V(0), V(1)});
+  sub.push_back({V(1), V(2)});
+  sub.push_back({V(1), V(3)});
+  sub.push_back({V(2), V(3)});
+  sub.push_back({V(3), V(4)});
+
+  IdTable leftOpTable(2, makeAllocator());
+  leftOpTable.push_back({V(0), V(1)});
+  leftOpTable.push_back({V(0), V(2)});
+  leftOpTable.push_back({V(0), V(3)});
+
+  IdTable expected(3, makeAllocator());
+  expected.push_back({V(1), V(4), V(0)});
+  expected.push_back({V(2), V(4), V(0)});
+  expected.push_back({V(3), V(4), V(0)});
+
+  TransitivePathSide left(std::nullopt, 0, Variable{"?start"}, 0);
+  TransitivePathSide right(std::nullopt, 1, V(4), 1);
+  auto T = makePathLeftBound(
+      std::move(sub), {Variable{"?start"}, Variable{"?target"}},
+      std::move(leftOpTable), 1, {Variable{"?x"}, Variable{"?start"}},
+      std::move(left), std::move(right), 0, std::numeric_limits<size_t>::max());
+
+  auto resultTable = T->computeResultOnlyForTesting();
+  assertSameUnorderedContent(expected, resultTable.idTable());
+}
+
+TEST_P(TransitivePathTest, idToRightBound) {
+  IdTable sub(2, makeAllocator());
+  sub.push_back({V(0), V(1)});
+  sub.push_back({V(1), V(2)});
+  sub.push_back({V(1), V(3)});
+  sub.push_back({V(2), V(3)});
+  sub.push_back({V(3), V(4)});
+
```

Comment:
You please refactor the creation of the test inputs to use the `makeIdTableFromVector` helper functions from the tests (in the whole file, modernization).